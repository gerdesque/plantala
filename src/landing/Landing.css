.app-landing {
  width: 40vmin;
  height: auto;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .app-landing {
    animation: app-landing-spin infinite 20s linear;
  }
}

@keyframes app-landing-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
