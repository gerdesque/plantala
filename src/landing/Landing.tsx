import React from 'react';
import { Grid } from '@material-ui/core';
import logo from '../assets/logo.webp';
import './Landing.css';

export default function Landing() {
  return (
    <Grid className="app-landing-container" container justify="center">
      <img src={logo} className="app-landing" alt="Plantala" />
    </Grid>
  );
}
