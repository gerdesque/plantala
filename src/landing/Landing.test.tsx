import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import { ShallowWrapper } from 'enzyme';
import Landing from './Landing';

describe('Landing', () => {

  let shallow: any;
  let wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

  beforeEach(() => {
    shallow = createShallow();
    wrapper = shallow(<Landing />);
  });

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render a logo', () => {
    expect(wrapper.find('img')).toHaveLength(1);
  });

});
