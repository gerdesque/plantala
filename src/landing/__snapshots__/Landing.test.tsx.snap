// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing should render correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  justify="center"
>
  <img
    alt="Plantala"
    className="app-landing"
    src="logo.png"
  />
</WithStyles(ForwardRef(Grid))>
`;
